### DISCUSSION TITLE
Update Nodesets automaticly

### DISCUSSION THREAD

**Discussion by swamper123** [FreeOpcUa/opcua-asyncio#424] [Ideas]
Hello there,

since we got some seperate files to update the UA Nodeset, we may add a method/create a single executable file to update everything at once?

This would include executing all generate_*.py files in schemas and an overwriting of the UA-Nodeset(-master) folder via git clone of the Nodesets master repo.

I could imagine either to put sth. like thtat into utils.py or a new file in the schemas folder (and the generate_*.py files in a subfolder perhaps?)?!

Since these Nodesets will be more and more, the Versions will be more and more important and I can imagine, that being uptodate in a simple way, may be a nice feature.

**Comment by swamper123** [comment]
Could we add the UA-Nodeset as a subrepo, so that we just make a git pull and the Nodesets would be automaticly there?
And then use a "generate_basic_nodeset.py" script to update?

**Reply by oroulet** [reply]
I have bad experience with subrepos. and there is little reason to have such a hard link here I think. We can make a script cloning the right repository though...

**Reply by swamper123** [reply]
Okay. Never had to deal with such a case.

So the "Refresh"-script shall do:
- git clone UA-Nodeset repo
- perhaps delete all files of that clone, except UA-Nodeset
- execute the genreate_*.py files.

I know that github actions provide some features, like a schedule thing...don't know if we could use such an action to be uptodate with UA-Nodeset (so that just the script would be executed and merged into master or into a branch until we approve it)?!

**Comment by swamper123** [comment]
I made something in #435 to make life a bit easier.

It would be cool, if that script could be triggered when changes in UA-Nodeset are made and the changes merged into master (or temp branch and merge them manually)....but I have no clue if github offers something like a github action subscription service?!
